mlflow/pyspark/ml/__init__.py

```diff
@@ -215,6 +222,12 @@ def _gen_stage_hierarchy_recursively(stage, uid_to_indexed_name_map):
                 tuned_estimator, uid_to_indexed_name_map
             ),
         }
+    elif any(_get_params_param_map(stage)):
+        sub_params = []
+        for _, param_value in _get_params_param_map(stage).items():
+            sub_hierarchy = _gen_stage_hierarchy_recursively(param_value, uid_to_indexed_name_map)
+            sub_params.append(sub_hierarchy)
+        return {"name": stage_name, "params": sub_params}
```

Comment:
why not make "sub_params" to be a dict (param_name --> param_value) ? otherwise it lost the param name information.